spec/syntax.md

```diff
@@ -91,159 +91,198 @@ The syntax specification takes into account the following design restrictions:
+>The same message defined in a `.properties` file:
+>
+>>```properties
+>>app.greetings.hello = {Hello, world!}
+>>```
 
-```js
-let hello = new MessageFormat('{Hello, world!}')
-hello.format()
-```
+>The same message defined inline in JavaScript:
+>
+>>```js
+>>let hello = new MessageFormat('{Hello, world!}')
+>>hello.format()
+>>```
 
 ### Expression
 
-An _expression_ represents a part of a message that will be determined
-during the message's formatting.
+An **_expression_** is a part of a _message_ that will be determined
+during the _message_'s formatting.
 
-An _expression_ always uses `{…}` delimiters.
-An _expression_ can appear as a local variable value, as a _selector_, and within a _pattern_.
+A **_placeholder_** is an _expression_ that appears inside of a _pattern_
+and which will be replaced during the formatting of the _message_.
 
-A simple _expression_ is a bare variable name:
+An _expression_ begins with U+007B LEFT CURLY BRACKET `{` 
+and ends with U+007D RIGHT CURLY BRACKET `}`.
 
-    {Hello, {$userName}!}
+An _expression_ can appear as a local variable value, as a _selector_, and within a _pattern_.
 
-### Formatting Functions
+> A simple _expression_ is a bare variable name:
+>
+>>```
+>>    {Hello, {$userName}!}
+>>```
 
-A _function_ is named functionality, possibly with _options_, that format,
-process, or operate on an _operand_ which may be either a _literal_ or a _variable_.
+### Function
 
-For example, a _message_ with a `$date` _variable_ formatted with the `:datetime` _function_:
+A **_function_** is a named modifier in an _expression_.
+A _function_ MAY be followed by zero or more _options_
 
-    {Today is {$date :datetime weekday=long}.}
+>For example, a _message_ with a `$date` _variable_ formatted with the `:datetime` _function_:
+>
+>>```
+>>    {Today is {$date :datetime weekday=long}.}
+>>```
 
-A _message_ with a `$userName` _variable_ formatted with
-the custom `:person` _function_ capable of
-declension (using either a fixed dictionary, algorithmic declension, ML, etc.):
+>A _message_ with a `$userName` _variable_ formatted with
+>the custom `:person` _function_ capable of
+>declension (using either a fixed dictionary, algorithmic declension, ML, etc.):
+>
+>>```
+>>    {Hello, {$userName :person case=vocative}!}
+>>```
 
-    {Hello, {$userName :person case=vocative}!}
+>A _message_ with a `$userObj` _variable_ formatted with
+>the custom `:person` _function_ capable of
+>plucking the first name from the object representing a person:
+>
+>>```
+>>    {Hello, {$userObj :person firstName=long}!}
+>>```
 
-A _message_ with a `$userObj` _variable_ formatted with
-the custom `:person` _function_ capable of
-plucking the first name from the object representing a person:
+_Functions_ can be _standalone_, or can be an _opening element_ or _closing element_.
 
-    {Hello, {$userObj :person firstName=long}!}
+A **_standalone_** _function_ is not expected to be paired with another _function_.
```

Comment:
The `standalone`, `open`, `close` are properties of the expression, not the function.

Reply:
Is it _expression_ or _placeholder_? That is, does it have meaning to say that this is an "open expression":

```
let $foo = {$bar +open}
```

Reply:
... also we should say that `standalone` is the default for expressions such as `{$foo}` that have only an operand.

Reply:
> Is it _expression_ or _placeholder_? That is, does it have meaning to say that this is an "open expression":
> 
> ```
> let $foo = {$bar +open}
> ```

We have not explicitly discussed this (yet!), but yes, that's what we have right now.

> ... also we should say that `standalone` is the default for expressions such as `{$foo}` that have only an operand.

I'm pretty sure that such a default is not currently defined, and that this is probably best considered within the context of discussing our function registry and our implementations' behaviour assigning function handlers to expressions that do not include an annotattion.

This PR should not make any such statements.